include/sqlpp11/sqlite3/connection.h

```diff
@@ -372,14 +372,29 @@ namespace sqlpp
         return static_cast<size_t>(sqlite3_changes(native_handle()));
       }
 
-      template <
-          typename Execute,
-          typename Enable = typename std::enable_if<not std::is_convertible<Execute, std::string>::value, void>::type>
-      size_t execute(const Execute& x)
-      {
-        _context_t context{*this};
-        serialize(x, context);
-        return execute(context.str());
+      template <
+          typename Execute,
+          typename std::enable_if<not std::is_convertible<Execute, std::string>::value 
+                                  and not sqlpp::is_prepared_statement_t<Execute>::value, int>::type = 0>
+      size_t execute(const Execute& x)
+      {
+        static_assert(not sqlpp::is_select_t<Execute>::value, "argument must not be a select statement - use operator() instead");

```

Comment:
Wouldn't `SELECT FOR UPDATE` make sense to execute?

Reply:
I never used this so did not think of it. However, as it is a select that does return rows - even if as an extra function it also locks them - I do not think `execute()` should be used to running them. See my comment/question above on differentiation of `execute()` from `query()`.

Furthermore, AFAIK SQLite does not support this syntax. SQLite locks the whole database file for a write so it would not make much sense anyway.